servers/services/src/test/java/org/projectnessie/services/impl/BaseTestServiceImpl.java

```diff
@@ -0,0 +1,525 @@
+    return commit(targetBranch.getName(), targetBranch.getHash(), meta, operations);
+  }
+
+  protected CommitResponse commit(
+      String branch, String expectedHash, CommitMeta meta, Operation... operations)
+      throws NessieConflictException, NessieNotFoundException {
+    Operations ops =
+        ImmutableOperations.builder().addOperations(operations).commitMeta(meta).build();
+    return treeApi().commitMultipleOperations(branch, expectedHash, ops);
+  }
+
+  protected Map<ContentKey, Content> contents(Reference reference, ContentKey... keys)
+      throws NessieNotFoundException {
+    return contents(reference.getName(), reference.getHash(), keys);
+  }
+
+  protected Map<ContentKey, Content> contents(String refName, String hashOnRef, ContentKey... keys)
+      throws NessieNotFoundException {
+    return contentApi()
+        .getMultipleContents(refName, hashOnRef, Arrays.asList(keys))
+        .getContents()
+        .stream()
+        .collect(Collectors.toMap(ContentWithKey::getKey, ContentWithKey::getContent));
+  }
+
+  protected String createCommits(
+      Reference branch, int numAuthors, int commitsPerAuthor, String currentHash)
+      throws NessieConflictException, NessieNotFoundException {
+    for (int j = 0; j < numAuthors; j++) {
+      String author = "author-" + j;
+      for (int i = 0; i < commitsPerAuthor; i++) {
+        ContentKey key = ContentKey.of("table" + i);
+
+        Put op;
+        try {
+          Content existing = contentApi().getContent(key, branch.getName(), currentHash);
+          op =
+              Put.of(
+                  key,
+                  IcebergTable.of("some-file-" + i, 42, 42, 42, 42, existing.getId()),
+                  existing);
+        } catch (NessieContentNotFoundException notFound) {
+          op = Put.of(key, IcebergTable.of("some-file-" + i, 42, 42, 42, 42));
+        }
+
+        String nextHash =
+            commit(
+                    branch.getName(),
+                    currentHash,
+                    CommitMeta.builder()
+                        .author(author)
+                        .message("committed-by-" + author)
+                        .properties(ImmutableMap.of("prop1", "val1", "prop2", "val2"))
+                        .build(),
+                    op)
+                .getTargetBranch()
+                .getHash();
+
+        assertThat(currentHash).isNotEqualTo(nextHash);
+        currentHash = nextHash;
+      }
+    }
+    return currentHash;
+  }
+
+  protected static final class ListPagedResponseHandler<E>
+      implements PagedResponseHandler<List<E>, E> {
+
+    final List<E> result = new ArrayList<>();
+
+    @Override
+    public List<E> build() {
+      return result;
+    }
+
+    @Override
+    public boolean addEntry(E entry) {
+      result.add(entry);
+      return true;
+    }
+
+    @Override
+    public void hasMore(String pagingToken) {
+      throw new UnsupportedOperationException();
```

Comment:
nit: rename this inner class to `UnlimitedListResponseHandler` for the sake of clarity?